synapse/util/async_helpers.py

```diff
@@ -309,12 +309,12 @@ def gather_results(  # type: ignore[misc]
     return deferred.addCallback(tuple)
 
 
-@attr.s(slots=True)
+@attr.s(slots=True, auto_attribs=True)
 class _LinearizerEntry:
     # The number of things executing.
-    count = attr.ib(type=int)
+    count: int
     # Deferreds for the things blocked from executing.
-    deferreds = attr.ib(type=collections.OrderedDict)
+    deferreds: collections.OrderedDict
```

Comment:
I wonder if this could just be `dict` now we're on 3.7+? Probably not worth changing.

Reply:
I think keeping it as `OrderedDict` is clearer for now!